releasenotes/notes/add-qpy-version-flag-6bb1756e671fde55.yaml

```diff
@@ -0,0 +1,15 @@
+---
+features:
+  - |
+    Added a new flag, ``version``, to the :func:`.qpy.dump` function which
+    optionally takes an integer value for the :ref:`qpy_format` version to
+    emit from the dump function. This is useful if you need to generate a QPY
+    file that will be loaded by an older version of Qiskit. However, the
+    supported versions to emit are very limited, only the latest QPY version
+    (which is the default), and the compatibility QPY version
+    which is :ref:`qpy_version_10` (which was introduced in Qiskit 0.45.0).
+    The compatibility version will remain fixed for the the entire 1.x.y major
+    version release series. This does not change the backwards compatibility
+    guarantees of the QPY format when calling :func:`.qpy.load`, it just enables
+    users to emit an older version of QPY to maintain compatibility and
+    interoperability between the 0.x and 1.x release series.
```

Comment:
This is probably the safest choice of compatibility mode.  I think this is _probably_ the right call, though part of me wants to argue in favour of allowing us to bump default version with minor releases, provided the minimum target version stays the same for a full major at least.

Reply:
Oh, that was my plan. If we add version 11 in this release `dump()` will emit 11 by default and if we introduced version 12 in 1.1.0 then dump() would default to 12 in 1.1.0 with the user option to roll it back to 10 using `version=10`. I feel the leading edge makes more sense here because of how tightly coupled QPY is to the circuit class changes holding back to version 10 by default would prevent us from keeping things compatible with later additions (or fixes). The wording here is fairly awkward because the compatibility version and the latest version are still both 10. This really only becomes clear when we introduce version 11.

Reply:
Ah ok, yeah, I can go with that too, provided we teach providers to set this flag when using QPY.